### PR TITLE
feat: add color for showMessage

### PR DESCRIPTION
**Pull request by nliu-ms**
image

### CODE REVIEW

File: packages/vscode-extension/src/qm/vsc_ui.ts [hunk 1]
```diff
@@ -493,27 +494,44 @@ export class VsCodeUI implements UserInteraction {
...
+  ): Promise<Result<string|undefined,FxError>>;
+
+  public async showMessage(
+    level: "info" | "warn" | "error",
+    message: Array<{content: string, color: Colors}>,
+    modal: boolean,
+    ...items: string[]
+  ): Promise<Result<string|undefined,FxError>>;
+
+  async showMessage(
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/qm/vsc_ui.ts:511]: Do we still need the above two functions?

**Review by chagong — APPROVED** [review]

File: packages/api/src/qm/ui.ts [hunk 2]
```diff
@@ -96,6 +97,16 @@ export interface UserInteraction {
     modal: boolean,
     ...items: string[]
   ): Promise<Result<string|undefined,FxError>>;
+  /**
+   * 
+   * @param message Only INFO level supports message color, color for other log levels will be ignored
```
Comment (HuihuiWu-Microsoft) [inline packages/api/src/qm/ui.ts:102]: Color only works for CLI?

**Review by jayzhang — APPROVED** [review]

File: packages/vscode-extension/src/qm/vsc_ui.ts (hunk 1 above)
Comment (nliu-ms) [reply packages/vscode-extension/src/qm/vsc_ui.ts:511]: This is the way to overload function, as chagong suggested.
image

File: packages/api/src/qm/ui.ts (hunk 2 above)
Comment (nliu-ms) [reply packages/api/src/qm/ui.ts]: comments updated.

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Pull request merged**